### PR TITLE
[ENHANCEMENT] Serialize PySpark DataFrame by converting to dictionary

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- added code to `convert_to_json_serializable()` and `ensure_json_serializable()` to handle pyspark DFs
- added single test (draft only)

### CODE REVIEW

File: great_expectations/core/util.py [hunk 1]
```diff
@@ -63,6 +63,14 @@ def convert_to_json_serializable(data):
     import numpy as np
     import pandas as pd
 
+    try:
+        import pyspark
+    except ImportError:
+        pyspark = None
+        logger.debug(
+            "Unable to load pyspark and pyspark.sql; install optional spark dependency for support."
```
Comment (alexsherstinsky) [inline great_expectations/core/util.py]: @Shinnnyshinshin Minor: The message leads one to believe that we are trying to import `pyspark.sql`, but it is not being imported (only `pyspark` on line 67).

File: great_expectations/core/util.py [hunk 2]
```diff
@@ -140,6 +148,13 @@ def convert_to_json_serializable(data):
     elif isinstance(data, pd.DataFrame):
         return convert_to_json_serializable(data.to_dict(orient="records"))
 
+    elif pyspark and isinstance(data, pyspark.sql.DataFrame):
```
Comment (alexsherstinsky) [inline great_expectations/core/util.py:149]: Why not just `from pyspark.sql import DataFrame` and then not check for `pyspark` in line 151 and just say `isinstance(data, DataFrame)`?..

File: great_expectations/core/util.py [hunk 3]
```diff
@@ -173,6 +188,14 @@ def ensure_json_serializable(data):
...
     import pandas as pd
 
+    try:
+        import pyspark
+    except ImportError:
+        pyspark = None
+        logger.debug(
+            "Unable to load pyspark and pyspark.sql; install optional spark dependency for support."
+        )
+
```
Comment (alexsherstinsky) [inline great_expectations/core/util.py:183]: Are we having a duplicate of the `import pyspark` section?  Lines 191 -- 198 seem to be the same as lines 66 -- 72...

File: great_expectations/core/util.py [hunk 4]
```diff
@@ -243,6 +266,14 @@ def ensure_json_serializable(data):
...
         ]
         return
+
+    elif pyspark and isinstance(data, pyspark.sql.DataFrame):
+        # using StackOverflow suggestion for converting pyspark df into dictionary
+        # https://stackoverflow.com/questions/43679880/pyspark-dataframe-to-dictionary-columns-as-keys-and-list-of-column-values-ad-di
+        return ensure_json_serializable(
+            dict(zip(data.schema.names, zip(*data.collect())))
+        )
+
```
Comment (alexsherstinsky) [inline great_expectations/core/util.py:261]: This looks same as in lines 152 -- 157...

File: great_expectations/core/util.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/core/util.py:149]: Oh nevermind -- we use both Pandas and Spark here...  OK; maybe leave this as is just update the error message to not talk about `pyspark.sql` unless this was on purpose.  Thanks.

File: tests/core/test_serialization.py [hunk 5]
```diff
@@ -46,3 +46,16 @@ def test_lossy_conversion():
 
     d = Decimal("0.1")
     assert not requires_lossy_conversion(d)
+
+
+# TODO add unittests for convert_to_json_serializable() and ensure_json_serializable()
+def test_serialization_of_spark_df(spark_session):
```
Comment (alexsherstinsky) [inline tests/core/test_serialization.py:52]: Beautiful!

**Review by alexsherstinsky — APPROVED** [review]
LGTM as long as the test pass (and address the comments -- if warranted).  Thanks so much!

File: great_expectations/core/util.py (hunk 4 above)
Comment (Shinnnyshinshin) [reply great_expectations/core/util.py:261]: the function calls are slightly different here `ensure_json_serializable()` vs `convert_to_json_serializable()`, but yes the inner function-call is the same. Would you prefer to have this separated out?

File: great_expectations/core/util.py (hunk 3 above)
Comment (Shinnnyshinshin) [reply great_expectations/core/util.py:183]: good point. I was following the existing pattern for pandas and numpy imports, but I think they would be much more efficient if it was moved to the top

File: great_expectations/core/util.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/core/util.py]: wonderful catch. made the change

File: great_expectations/core/util.py (hunk 2 above)
Comment (Shinnnyshinshin) [reply great_expectations/core/util.py:149]: done!

**Pull request merged**